src/dynamics/omega_limit.lean

```diff
@@ -0,0 +1,313 @@
+/-
+Copyright (c) 2020 Jean Lo. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jean Lo
+-/
+import dynamics.flow
+
+/-!
+# ω-limits
+
+For a function `ϕ : τ → α → β` where `β` is a topological space, we
+define the ω-limit under `ϕ` of a set `s` in `α` with respect to
+filter `f` on `τ`: an element `y : β` is in the ω-limit of `s` if the
+forward images of `s` intersect arbitrarily small neighbourhoods of
+`y` frequently "in the direction of `f`".
+
+In practice `ϕ` is often a continuous monoid-act, but the definition
+requires only that `ϕ` has a coercion to the appropriate function
+type. In the case where `τ` is `ℕ` or `ℝ` and `f` is `at_top`, we
+recover the usual definition of the ω-limit set as the set of all `y`
+such that there exist sequences `(tₙ)`, `(xₙ)` such that `ϕ tₙ xₙ ⟶ y`
+as `n ⟶ ∞`.
+
+## Notations
+
+The `omega_limit` locale provides the localised notation `ω` for
+`omega_limit`, as well as `ω⁺` and `ω⁻` for `omega_limit at_top` and
+`omega_limit at_bot` respectively for when the acting monoid is
+endowed with an order.
+-/
+
+open set function filter
+
+/-!
+### ω-limit sets
+-/
+
+section omega_limit
+
+variables {τ : Type*} {α : Type*} {β : Type*}
+
+/-- The ω-limit of a set `s` under `ϕ` with respect to a filter `f` is
+    ⋂ u ∈ f, cl (ϕ u s). -/
+def omega_limit [topological_space β] (f : filter τ) (ϕ : τ → α → β) (s : set α) : set β :=
+⋂ u ∈ f, closure (image2 ϕ u s)
+
+localized "notation `ω` := omega_limit" in omega_limit
+
+localized "notation `ω⁺` := omega_limit at_top" in omega_limit
+localized "notation `ω⁺` := omega_limit at_bot" in omega_limit
+
+variables [topological_space β]
+variables (f : filter τ) (ϕ : τ → α → β) (s s₁ s₂: set α)
+
+-- some elementary properties:
```

Comment:
Here and below: making these line comments into module doc strings will let folks browsing the code online see these helpful notes as well. You can either make them plain text,
```suggestion
/-!
some elementary properties: 
-/
```
or add additional heading levels:
```suggestion
/-!
#### some elementary properties
-/
```

Reply:
this has been done in a3f38746704e67a3b0eef108627c3a875069665c , which also adds some more docstrings.